modules/configuration/src/main/java/org/apache/ignite/internal/configuration/asm/ConfigurationAsmGenerator.java

```diff
@@ -588,83 +621,97 @@ private void addNodeChangeMethod(
     }
 
     /**
-     * Implements {@link InnerNode#traverseChildren(ConfigurationVisitor)} method.
+     * Implements {@link InnerNode#traverseChildren(ConfigurationVisitor, boolean)} method.
+     *
      * @param classDef Class definition.
-     * @param schemaFields Array of all schema fields that represent configurations.
+     * @param schemaFields Fields of the schema and its extensions.
      * @param fieldDefs Definitions for all fields in {@code schemaFields}.
+     * @param schemaExtensions Internal extensions of the configuration schema.
      */
     private void addNodeTraverseChildrenMethod(
         ClassDefinition classDef,
-        Field[] schemaFields,
-        Map<String, FieldDefinition> fieldDefs
+        Set<Field> schemaFields,
```

Comment:
Can I safely assume that this method could have parameters like "schemaFields" and "extensionsFields" and you decided to merge them together, only to split them back later? This definitely makes the code more complicated